src/services/actions/auth.js

```diff
@@ -0,0 +1,284 @@
+    type: REGISTER_SUCCESS,
+    payload,
+  };
+};
+
+export const registerFailed = () => {
+  return {
+    type: REGISTER_FAILED,
+  };
+};
+
+export const loginRequest = () => {
+  return {
+    type: LOGIN_REQUEST,
+  };
+};
+
+export const loginSuccess = (payload) => {
+  return {
+    type: LOGIN_SUCCESS,
+    payload,
+  };
+};
+
+export const loginFailed = () => {
+  return {
+    type: LOGIN_FAILED,
+  };
+};
+
+export const logoutRequest = () => {
+  return {
+    type: LOGOUT_REQUEST,
+  };
+};
+
+export const logoutSuccess = (payload) => {
+  return {
+    type: LOGOUT_SUCCESS,
+    payload,
+  };
+};
+
+export const logoutFailed = () => {
+  return {
+    type: LOGOUT_FAILED,
+  };
+};
+
+export const userRequest = () => {
+  return {
+    type: USER_REQUEST,
+  };
+};
+
+export const userSuccess = (payload) => {
+  return {
+    type: USER_SUCCESS,
+    payload,
+  };
+};
+
+export const userFailed = () => {
+  return {
+    type: USER_FAILED,
+  };
+};
+
+export const updateUserRequest = () => {
+  return {
+    type: UPDATE_USER_REQUEST,
+  };
+};
+
+export const updateUserSuccess = (payload) => {
+  return {
+    type: UPDATE_USER_SUCCESS,
+    payload,
+  };
+};
+
+export const updateUserFailed = () => {
+  return {
+    type: UPDATE_USER_FAILED,
+  };
+};
+
+export const forgotPasswordReq = () => {
+  return {
+    type: FORGOT_PASSWORD_REQUEST,
+  };
+};
+
+export const forgotPasswordSuccess = (payload) => {
+  return {
+    type: FORGOT_PASSWORD_SUCCESS,
+    payload,
+  };
+};
+
+export const forgotPasswordFailed = () => {
+  return {
+    type: FORGOT_PASSWORD_FAILED,
+  };
+};
+
+export const resetPasswordReq = () => {
+  return {
+    type: RESET_PASSWORD_REQUEST,
+  };
+};
+
+export const resetPasswordSuccess = (payload) => {
+  return {
+    type: RESET_PASSWORD_SUCCESS,
+    payload,
+  };
+};
+
+export const resetPasswordFailed = () => {
+  return {
+    type: RESET_PASSWORD_FAILED,
+  };
+};
+
+export const registerUser = (data) => (dispatch) => {
+  dispatch(registerRequest());
+  signUpUserRequest(data)
+    .then((res) => {
+      if (res && res.success) {
+        const authToken = res.accessToken.split("Bearer ")[1];
+        setCookie("token", authToken);
+        localStorage.setItem("refreshToken", res.refreshToken);
+        dispatch(registerSuccess(res.user));
+      } else {
+        dispatch(registerFailed());
+      }
+    })
+    .catch((error) => {
+      dispatch(registerFailed());
+    });
+};
+
+export const loginUser = (data) => (dispatch) => {
+  dispatch(loginRequest());
+  signInUserRequest(data)
+    .then((res) => {
+      if (res && res.success) {
+        const authToken = res.accessToken.split("Bearer ")[1];
+        setCookie("token", authToken);
+        localStorage.setItem("refreshToken", res.refreshToken);
+        dispatch(loginSuccess(res.user));
+      } else {
+        dispatch(loginFailed());
+      }
+    })
+    .catch((error) => {
+      dispatch(loginFailed());
+    });
```

Comment:
Можно лучше:
тут чтобы не дублировать функциональность, связанную с ошибкой при запросе, в блоке else можно выбрасывать ошибку, и она автоматически попадет в блок catch:
```suggestion
      } 
        throw new Error('Some text');
    })
    .catch((error) => {
      dispatch(loginFailed());
    });
```